pkg/init/cmd/service/runc.go

```diff
@@ -52,8 +58,32 @@ func runcInit(rootPath string) int {
 		}
 		pidfile := filepath.Join(tmpdir, name)
 		cmd := exec.Command(runcBinary, "create", "--bundle", path, "--pid-file", pidfile, name)
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
+
+		// stream stdout and stderr to respective files
+		// ideally we want to use io.MultiWriter here, sending one stream to stdout/stderr, another to the files
+		// however, this hangs if we do, due to a runc bug, see https://github.com/opencontainers/runc/issues/1721#issuecomment-366315563
+		// once that is fixed, this can be cleaned up
+		stdoutFile := filepath.Join(logDir, serviceType+"."+name+".out.log")
+		stdout, err := os.OpenFile(stdoutFile, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0644)
+		if err != nil {
+			log.Printf("Error opening stdout log file: %v", err)
+			status = 1
+			continue
+		}
+		defer stdout.Close()
+
+		stderrFile := filepath.Join(logDir, serviceType+"."+name+".err.log")
+		stderr, err := os.OpenFile(stderrFile, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0644)
```

Comment:
I don't know that we should append logs, as we cant rotate them. Normally they will be on tmpfs anyway so it won't matter.

Actually wondering if we should write the onboot logs to `/tmp` anyway, as often `/var` will be overmounted so these won't be any use.

Reply:
> I don't know that we should append logs, as we cant rotate them

Do you mean we should remove `|os.O_APPEND`?

> wondering if we should write the onboot logs to /tmp anyway, as often /var will be overmounted so these won't be any use

services logs are in `/var/log`; do we not want them in the same place? If there is a concern that a `mount` container in `onboot` might overwrite it, then be explicit about ordering?

Reply:
Yes but `onboot` is often there to mount a new `/var`, that is all done by the time services start, but for `onboot` part way through the containers it may get overmounted and then you lose all the earlier logs.

(you can just use `os.Create` I think if you dont want to append)

Reply:
Ok. I’ll switch it to remove `os.Apoend`

Logs in `/tmp` seem strange. Is there another place?